packages/paymaster/src/BiconomyPaymaster.ts

```diff
@@ -1,49 +1,246 @@
+                smartAccountInfo: smartAccountInfo
+              }
+            }
+          ], // As per current API
+          id: getTimestampInSeconds(),
+          jsonrpc: '2.0'
+        }
+      })
+
+      if (response && response.result) {
+        Logger.log('feeInfo ', response.result)
+        if (response.result.mode == PaymasterMode.ERC20) {
+          const feeQuotesResponse: Array<PaymasterFeeQuote> = response.result.feeQuotes
+          const paymasterAddress: string = response.result.paymasterAddress
+          // check all objects iterate and populate below calculation for all tokens
+          return { feeQuotes: feeQuotesResponse, tokenPaymasterAddress: paymasterAddress }
+        } else if (response.result.mode == PaymasterMode.SPONSORED) {
+          const paymasterAndData: string = response.result.paymasterAndData
+          return paymasterAndData
+        } else {
+          throw new Error('Failed to fetch feeQuote or paymaster data')
+        }
+      } else {
+        // Review if any case response.ok could be true and response would be below
+        /*{
+          "statusCode": 500,
+          "message": "Internal server error"
+        }*/
+        // Note: we may not throw if we include strictMode off and return paymasterData '0x'.
+        // pm service could handle this case by case. Needs Review
+        throw new Error('Failed to fetch feeQuote or paymaster data')
+      }
+    } catch (error: any) {
+      Logger.error("can't query fee quotes - reason: ", error)
+      // Note: we may not throw if we include strictMode off and return paymasterData '0x'.
+      throw new Error('Failed to fetch feeQuote or paymaster data' + error.toString())
+    }
+  }
+
+  async getPaymasterAndData(
+    userOp: Partial<UserOperation>,
+    paymasterServiceData?: SponsorUserOperationDto // mode is necessary. partial context of token paymaster or verifying
+  ): Promise<string> {
     try {
       userOp = await resolveProperties(userOp)
-      userOp.nonce = Number(userOp.nonce)
-      userOp.callGasLimit = Number(userOp.callGasLimit)
-      userOp.verificationGasLimit = Number(userOp.verificationGasLimit)
-      userOp.maxFeePerGas = Number(userOp.maxFeePerGas)
-      userOp.maxPriorityFeePerGas = Number(userOp.maxPriorityFeePerGas)
-      userOp.preVerificationGas = Number(userOp.preVerificationGas)
+      userOp.nonce = BigNumber.from(userOp.nonce).toHexString()
+      userOp.callGasLimit = BigNumber.from(userOp.callGasLimit).toHexString()
+      userOp.verificationGasLimit = BigNumber.from(userOp.verificationGasLimit).toHexString()
+      userOp.maxFeePerGas = BigNumber.from(userOp.maxFeePerGas).toHexString()
+      userOp.maxPriorityFeePerGas = BigNumber.from(userOp.maxPriorityFeePerGas).toHexString()
+      userOp.preVerificationGas = BigNumber.from(userOp.preVerificationGas).toHexString()
       userOp.signature = '0x'
       userOp.paymasterAndData = '0x'
```

Comment:
Can be extracted into a different method as mentioned in earlier comments.

Reply:
done